src/main/java/core/basesyntax/services/impl/DataProcessorServiceImpl.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax.services.impl;
+
+import core.basesyntax.exception.InvalidDataException;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.services.DataProcessorService;
+import java.util.List;
+
+public class DataProcessorServiceImpl implements DataProcessorService {
+    private static final String SPLIT_DELIMITER = ",";
+    private static final int OPERATION_TYPE_INDEX = 0;
+    private static final int FRUIT_TYPE_INDEX = 1;
+    private static final int FRUIT_QUANTITY_INDEX = 2;
+
+    @Override
+    public List<FruitTransaction> processInputData(List<String> dataFromFile) {
+        return dataFromFile.stream()
+                .map(this::getFruitTransaction)
+                .toList();
+    }
+
+    private FruitTransaction getFruitTransaction(String data) {
+            String[] processedData = data.split(SPLIT_DELIMITER);
```

Comment:
Consider adding validation to check the length of `processedData` before accessing its elements to avoid potential `ArrayIndexOutOfBoundsException` if the input data is malformed.